oci/spec_opts.go

```diff
@@ -1264,3 +1273,32 @@ func WithDevShmSize(kb int64) SpecOpts {
 		return ErrNoShmMount
 	}
 }
+
+// freebsdLinuxEmulationMounts modifies the mount spec to mount emulated Linux filesystems on FreeBSD,
+// as per: https://wiki.freebsd.org/LinuxJails
+func freebsdLinuxEmulationMounts(s *Spec) error {
+	var mounts []specs.Mount
+	for _, m := range s.Mounts {
+		path := filepath.Clean(m.Destination)
+		if path == "/proc" {
+			mounts = append(mounts, specs.Mount{
+				Destination: "/proc",
+				Type:        "linprocfs",
+				Source:      "linprocfs",
+				Options:     []string{},
+			})
+			continue
+		}
+		if path == "/dev/fd" {
+			m.Options = append(m.Options, "linrdlink")
+		}
+		mounts = append(mounts, m)
+	}
+	mounts = append(mounts, specs.Mount{
+		Destination: "/sys",
+		Type:        "linsysfs",
+		Source:      "linsysfs",
+		Options:     []string{}})
```

Comment:
Should we add `nosuid`, `noexec`, `ro` options to follow linux containers more closely? Same applies to procfs.

Reply:
I believe nosuid and noexec wouldn't be needed here as it's a special FS that wouldn't allow executing anyway, I believe it's the same for ro.
None of them should break mounting though, so they should be safe to add.

@emaste could you confirm my theory?

Reply:
@gizahNL I would agree.

I am not sure if linsysfs/linprocfs have any writable entries anyway, but ro seems sensible for consistency w/ Linux and potential future writable nodes even if there are none today. Likewise I think nosuid and noexec will have no effect, but also will cause no harm and are fine for consistency / clarity of intent.